src/transformers/configuration_encoder_decoder.py

```diff
@@ -0,0 +1,82 @@
+# coding=utf-8
+# Copyright 2018 The Google AI Language Team Authors and The HuggingFace Inc. team.
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" BERT model configuration """
+
+
+import copy
+import logging
+
+from .configuration_auto import AutoConfig
+from .configuration_utils import PretrainedConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+class EncoderDecoderConfig(PretrainedConfig):
+    r"""
+        This is the configuration class to store the configuration of a :class:`~transformers.EncoderDecoderModel`.
+        It is used to instantiate an Encoder Decoder model according to the specified arguments, defining the encoder and decoder configs.
+
+        Configuration objects inherit from  :class:`~transformers.PretrainedConfig` and can be used
+        to control the model outputs. Read the documentation from  :class:`~transformers.PretrainedConfig`
+        for more information.
+
+
+        Args:
+            kwargs: (`optional`) Remaining dictionary of keyword arguments. Notably:
+                encoder (:class:`PretrainedConfig`, optional, defaults to `None`):
+                    An instance of a configuration object that defines the encoder config.
+                encoder (:class:`PretrainedConfig`, optional, defaults to `None`):
+                    An instance of a configuration object that defines the decoder config.
+    """
+    model_type = "encoder_decoder"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        assert (
+            "encoder" in kwargs and "decoder" in kwargs
+        ), "Config has to be initialized with encoder and decoder config"
+        encoder_config = kwargs.pop("encoder")
+        encoder_model_type = encoder_config.pop("model_type")
+        decoder_config = kwargs.pop("decoder")
+        decoder_model_type = decoder_config.pop("model_type")
+        self.encoder = AutoConfig.for_model(encoder_model_type, **encoder_config)
+        self.decoder = AutoConfig.for_model(decoder_model_type, **decoder_config)
+        self.is_encoder_decoder = True
+
+    @classmethod
+    def from_encoder_decoder_pretrained(cls, encoder_config, decoder_config):
```

Comment:
`<nit>`

the other from_pretrained takes `pretrained_model_name_or_path: str`, this takes 2 configs.
I would name it `from_configs`...
From the name you might also guess the arguments are instances of models.
Also type hints would be nice here.
`</nit>`

Reply:
I understand the consistency argument though. Fair enough.

Reply:
Hmm, I see your point! For me it would be fine to rename it to `from_encoder_decoder_configs()` What do you think @thomwolf @LysandreJik @julien-c ?

Reply:
I'm neutral. Choose the option you like @patrickvonplaten

Reply:
Ok went for @sshleifer suggestion.